packages/flutter_tools/lib/src/linux/build_linux.dart

```diff
@@ -12,14 +13,42 @@ import '../convert.dart';
 import '../globals.dart';
 import '../project.dart';
 
+// The name of the icu data file.
+const String _kIcuDataName = 'icudtl.dat';
+
+/// The name of the engine library.
+const String _kLibraryName = 'libflutter_linux.so';
+
 /// Builds the Linux project through the Makefile.
 Future<void> buildLinux(LinuxProject linuxProject, BuildInfo buildInfo) async {
   /// Cache flutter root in linux directory.
   linuxProject.editableHostAppDirectory.childFile('.generated_flutter_root')
     ..createSync(recursive: true)
     ..writeAsStringSync(Cache.flutterRoot);
-
   final String buildFlag = buildInfo?.isDebug == true ? 'debug' : 'release';
+  final Directory cacheDirectory = fs.directory(artifacts.getEngineArtifactsPath(TargetPlatform.linux_x64));
+  final Directory buildDirectory = fs.directory(fs.path.join(getLinuxBuildDirectory(), 'cache', 'flutter_library'));
+  final File icuData = fs.file(fs.path.join(cacheDirectory.path, _kIcuDataName));
+  final File libraryFile = fs.file(fs.path.join(cacheDirectory.path, _kLibraryName));
+
+  // Copy the source files and headers.
+  copyDirectorySync(cacheDirectory, buildDirectory);
+
+  // Copy the ICU data.
+  final File icuDestination = fs.file(fs.path.join(getLinuxBuildDirectory(), buildFlag, 'data', _kIcuDataName));
```

Comment:
The copy targets not be directly in the build output directory. The `data` and `lib` directories are for the actual binary output location; the idea is that the makefile copies everything into those locations so that there's a directory that could be zipped up and moved until there's a real installer plan for Linux. But that's something that should be done by the Makefile on the app side (copying from `cache/flutter_library/`, or with the above change just `flutter/` to the bundle target location), not the `flutter_tool` side.